Src/Support/Google.Apis.Auth/OAuth2/IExternalAccountCredential.cs

```diff
@@ -0,0 +1,81 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Http;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Represents and external account credential. Used mostly for unifying different
+    /// external account credentials hierarchies.
+    /// </summary>
+    internal interface IExternalAccountCredential: IGoogleCredential,
+        IHttpExecuteInterceptor, IHttpUnsuccessfulResponseHandler
+    {
+        /// <summary>
+        /// The STS audience which contains the resource name for the
+        /// workload identity pool or the workforce pool
+        /// and the provider identifier in that pool.
+        /// </summary>
+        string Audience { get; }
+
+        /// <summary>
+        /// The STS subject token type based on the OAuth 2.0 token exchange spec.
+        /// </summary>
+        string SubjectTokenType { get; }
+
+        /// <summary>
+        /// The STS token exchange endpoint.
+        /// </summary>
+        string TokenUrl { get; }
+
+        /// <summary>
+        /// The URL for the service account impersonation request.
+        /// This is only required for workload identity pools when APIs to be accessed
+        /// have not integrated with UberMint. If this is not available,
+        /// the STS returned GCP access token is directly used.
+        /// </summary>
+        string ServiceAccountImpersonationUrl { get; }
+
+        /// <summary>
+        /// The endpoint used to retrieve the account related information (eg. email, username, uid, etc).
+        /// This is needed for gCloud session account identification.
+        /// </summary>
+        string TokenInfoUrl { get; }
+
+        /// <summary>
+        /// The Client ID.
+        /// </summary>
+        /// <remarks>
+        /// Client ID and Client secret are currently only required if the token info endpoint
+        /// needs to be called with the generated GCP access token.
+        /// When provided, STS will be called with additional basic authentication using
+        /// ClientId as username and ClientSecret as password.
+        /// </remarks>
+        string ClientId { get; }
```

Comment:
Does the client ID/secret have anything to do with the credential? AFAIK, we currently only support the gcloud ID here.